app/javascript/onboarding/__tests__/Onboarding.test.jsx

```diff
@@ -76,16 +76,21 @@ describe('<Onboarding />', () => {
   ]);
   const dataUser = JSON.stringify({
     followed_tag_names: ['javascript'],
+    profile_image_90: 'mock_url_link',
+    name: 'firstname lastname',
+    username: 'username',
   });
 
   describe('IntroSlide', () => {
     let onboardingSlides;
 
     beforeEach(() => {
-      onboardingSlides = initializeSlides(0);
+      onboardingSlides = initializeSlides(0, dataUser);
     });
 
     test('renders properly', () => {
+      // For some reason, this snapshot does not ever pick up on the dataUser properties
+      // so the snapshots do not render the user's name and render the fallback greeting instead.
```

Comment:
As this comment explains, the snapshot does not update with the user's name. I would _expect_ that [this line in the snapshot](https://github.com/thepracticaldev/dev.to/blob/5ae0a914f9409aa2a4792b999b88906f8b74ea10/app/javascript/onboarding/__tests__/__snapshots__/Onboarding.test.jsx.snap#L453) would read: `firstname lastname — welcome to DEV!`, but as you can see, it does not 😞 

I spent a lot of time trying to debug this today, and from what I can tell, the `userData` just doesn't seem to exist for this test. I honestly cannot explain why this happens in the testing environment (it works locally!), and why it happens only for this test (as @maestromac can attest, it works for him in https://github.com/thepracticaldev/dev.to/pull/7163).

I didn't want to spend too much more time debugging this, but if anyone sees what I'm doing wrong here, or has ideas on how to fix this, please do let me know. In the meantime, I'm adding this comment to help provide context to our future selves.

Reply:
UPDATE: thanks to @maestromac's fab debugging skills, this issue has been solved!! ✨